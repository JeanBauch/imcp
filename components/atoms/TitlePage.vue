<template>
	<div class="flex flex-col items-center gap-2 sm:gap-3 md:gap-4">
		<h1 :class="[themeColors.textColor, width]" class="text-4xl md:text-5xl xl:text-6xl font-semibold inline-block text-center xl:leading-tight">
			{{ title }}
		</h1>
		<div :class="themeColors.backgroundSeparetor" class="w-1/4 max-w-[4rem] h-1 rounded-full" />
	</div>
</template>

<script lang="ts">
import Vue from 'vue';

export default Vue.extend({
	data () {
		return {
			themeColors: {
				textColor: `text-${this.titleColor === 'blue' ? 'secondary' : 'primary'}-${this.titleColor}-500`,
				backgroundSeparetor: `bg-${this.separetorColor === 'blue' ? 'secondary' : 'primary'}-${this.separetorColor}-500`,
			},
		};
	},
	props: {
		title: {
			type: String,
			required: true,
		},
		titleColor: {
			type: String,
			default: 'orange',
			validator: (value: string) => [
				'yellow',
				'orange',
				'bordo',
				'beige',
				'blue',
			].includes(value.toLowerCase()),
		},
		separetorColor: {
			type: String,
			default: 'yellow',
			validator: (value: string) => [
				'yellow',
				'orange',
				'bordo',
				'beige',
				'blue',
			].includes(value.toLowerCase()),
		},
		width: {
			type: String,
			default: 'w-min',
		},
	},
});
</script>
