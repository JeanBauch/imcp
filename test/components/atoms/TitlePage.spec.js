import { shallowMount } from '@vue/test-utils';

import TitlePage from '~/components/atoms/TitlePage.vue';

describe('Components:atoms:ButtonInstrument', () => {
	const factory = (values = {}) => {
		return shallowMount(TitlePage, {
			propsData: {
				...values,
			},
		});
	};
	describe('Snapshots', () => {
		it('should render component with correctly props', () => {
			const wrapper = factory({ title: 'Noticias', titleColor: 'orange', separetorColor: 'yellow' });

			expect(wrapper.html()).toMatchSnapshot();
		});

		it('should render correctly data as with props datas', () => {
			const wrapper = factory({ title: 'Noticias', titleColor: 'orange', separetorColor: 'yellow' });

			expect(wrapper.find('.text-primary-orange-500').exists()).toBeTruthy();
			expect(wrapper.find('.bg-primary-yellow-500').exists()).toBeTruthy();
		});

		it('should render correctly data with color:orange', async () => {
			const wrapper = factory({ title: 'Noticias', titleColor: 'orange', separetorColor: 'yellow' });

			await wrapper.setData({ themeColors: { textColor: 'text-primary-orange-500' } });
			await wrapper.setData({ themeColors: { backgroundSeparetor: 'bg-primary-yellow-500' } });

			expect(wrapper.vm.themeColors.textColor).toBe('text-primary-orange-500');
			expect(wrapper.vm.themeColors.backgroundSeparetor).toBe('bg-primary-yellow-500');
		});

		it('should render correctly data with color:blue', async () => {
			const wrapper = factory({ title: 'Noticias', titleColor: 'blue', separetorColor: 'blue' });

			await wrapper.setData({ themeColors: { textColor: 'text-secondary-blue-500' } });
			await wrapper.setData({ themeColors: { backgroundSeparetor: 'bg-secondary-blue-500' } });

			expect(wrapper.vm.themeColors.textColor).toBe('text-secondary-blue-500');
			expect(wrapper.vm.themeColors.backgroundSeparetor).toBe('bg-secondary-blue-500');
		});
	});
});
