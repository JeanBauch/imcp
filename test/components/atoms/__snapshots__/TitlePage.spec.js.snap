// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components:atoms:ButtonInstrument Snapshots should render component with correctly props 1`] = `
"<div class=\\"flex flex-col items-center gap-2 sm:gap-3 md:gap-4\\">
  <h1 class=\\"text-4xl md:text-5xl xl:text-6xl font-semibold inline-block text-center xl:leading-tight text-primary-orange-500 w-min\\">
    Noticias
  </h1>
  <div class=\\"w-1/4 max-w-[4rem] h-1 rounded-full bg-primary-yellow-500\\"></div>
</div>"
`;
